src/scripts/api.ts

```diff
@@ -63,43 +48,52 @@ export const createRequest = ({attribute, geographicLevel, location, year, cropU
     (queryParams?.statisticcat_desc as ICropCategory)[cropUnits] :
     statisticcat_desc;
 
-  const locationHeader = geographicLevel === "REGION : MULTI-STATE" ? "region_desc" : geographicLevel === "County" ? "county_name" : "state_name";
-
-  const baseReq = `${baseURL}` +
-  `&sect_desc=${encodeURIComponent(sect_desc)}` +
-  `&group_desc=${encodeURIComponent(group_desc)}` +
-  `&commodity_desc=${encodeURIComponent(commodity_desc)}` +
-  `&statisticcat_desc=${encodeURIComponent(cat as string)}` +
-  `&domain_desc=${encodeURIComponent(domain_desc)}` +
-  `&agg_level_desc=${encodeURIComponent(geographicLevel)}` +
-  `&${locationHeader}=${encodeURIComponent(location)}` +
-  `&year=${year}`;
-
-  let req = baseReq;
-
-  item.forEach(subItem => {
-    req = req + `&short_desc=${encodeURIComponent(subItem)}`;
-  });
-
-  // hacky but we need to custom-build the req in this one case
-  if (attribute === "Total Farmers" && year !== "2017") {
+  let req = "";
+  if (attribute === "Total Farmers" && !years.includes("2017")) {
+    // we need to custom-build the req in this one case
     req = `${baseURL}` +
     `&sect_desc=${encodeURIComponent("DEMOGRAPHICS")}` +
     `&group_desc=${encodeURIComponent("OPERATORS")}` +
     `&commodity_desc=${encodeURIComponent("OPERATORS")}` +
     `&statisticcat_desc=${encodeURIComponent("OPERATORS")}` +
     `&domain_desc=${encodeURIComponent("TOTAL")}` +
     `&agg_level_desc=${encodeURIComponent(geographicLevel)}` +
-    `&${locationHeader}=${encodeURIComponent(location)}` +
-    `&year=${year}` +
     `&short_desc=${encodeURIComponent("OPERATORS, (ALL) - NUMBER OF OPERATORS")}`;
+  } else {
+    req = `${baseURL}` +
+    `&sect_desc=${encodeURIComponent(sect_desc)}` +
+    `&group_desc=${encodeURIComponent(group_desc)}` +
+    `&commodity_desc=${encodeURIComponent(commodity_desc)}` +
+    `&statisticcat_desc=${encodeURIComponent(cat as string)}` +
+    `&domain_desc=${encodeURIComponent(domain_desc)}` +
+    `&agg_level_desc=${encodeURIComponent(geographicLevel)}`;
   }
 
-  // if we are creating a request at the county level, we need to also pass in a state
-  if (state) {
-    req += `&state_name=${state}`;
+  years.forEach((year) => {
+    req = req + `&year=${year}`;
+  });
+
+  states.forEach((state) => {
+    if (geographicLevel === "REGION : MULTI-STATE") {
+      if (state !== "Alaska") {
+        const region = multiRegions.find((r) => r.States.includes(state));
+        req = req + `&region_desc=${encodeURIComponent(region?.Region as string)}`;
+      }
+    } else {
+      req = req + `&state_name=${encodeURIComponent(state)}`;
+
+    }
+  });
+
+  item.forEach(subItem => {
+    req = req + `&short_desc=${encodeURIComponent(subItem)}`;
+  });
+
+  if (geographicLevel === "REGION : MULTI-STATE") {
+    req = req + `&freq_desc=ANNUAL`;
```

Comment:
One note (but no need to change the code) - you could alternatively create a [SearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) object at the start of this code and then use its `append` method to add each parameter (`append` allows for multiples of the same key) and then use its `toString` method to get the query string with escaped values.  

Something like this:

```
const params = new SearchParams();
params.append("sect_desc", sect_desc);
params.append("group_desc", group_desc);
... more appends here ...
const req = `${baseUrl}?${params.toString()}`;
```